src/io/ExportVTK.cpp

```diff
@@ -145,6 +146,81 @@ void ExportVTK::exportData(
   }
 }
 
+void ExportVTK::exportGradients(
+    std::ofstream &   outFile,
+    const mesh::Mesh &mesh)
+{
+  for (const mesh::PtrData &data : mesh.data()) {
+    if (data->hasGradient()) {
+      Eigen::MatrixXd &values = data->gradientValues();
+      if (data->getDimensions() > 1) {
+        const int dim = data->getDimensions();
+        int       colm{0};
+        const int datasize = values.cols() / dim;
+
+        outFile << "VECTORS " << data->getName() << "_dx"
+                << " double\n";
+        for (int j = 0; j < datasize; j++) { // Over columns
+          int i = 0;
+          for (; i < dim; i++) {
```

Comment:
IIRC we always need to export 3D data, as vtk does not support 2D data. Usually, we set the third component just to zero. Please check this again.